src/Sema.zig

```diff
@@ -22885,6 +22885,17 @@ fn zirBuiltinExtern(
     if (!ty.isPtrAtRuntime()) {
         return sema.fail(block, ty_src, "expected (optional) pointer", .{});
     }
+    if (!try sema.validateExternType(ty, .other)) {
```

Comment:
I think this should check `ty.childType()` since `@extern` should return the same as taking the address of an extern variable (so `*[]u8` is not a valid type for it but `**[]u8` is).

Reply:
Why so? It can be slice and ```childType()``` will not take it into account

Reply:
`@extern(*[]u8)` should be an error because it is equivalent to `extern var foo: []u8` which is an error.